src/schemas/jsonSchema/loaders/cvc.js

```diff
@@ -0,0 +1,79 @@
+const { parseIdentifier } = require('../../../lib/stringUtils');
+const { services } = require('../../../services');
+
+const rootUri = 'http://identity.com/schemas/';
+
+class CVCSchemaLoader {
+  constructor(baseRemoteUri = undefined) {
+    this.baseRemoteUri = baseRemoteUri;
+  }
+
+  /**
+   * Gets the schema id based on the identifier
+   */
+  // eslint-disable-next-line class-methods-use-this
+  schemaId(identifier) {
+    return rootUri + identifier;
+  }
+
+  /**
+   * Tests to see if this loader is valid for the supplied identifier
+   */
+  // eslint-disable-next-line class-methods-use-this
+  valid(identifier) {
+    return /^(claim|credential|type)-(cvc|alt):.*$/.test(identifier) || /^cvc:.*$/.test(identifier);
+  }
+
+  /**
+   * Loads the schema based on the identifier
+   */
+  async loadSchema(identifier) {
+    let identifierPath;
+
+    if (/^cvc:.*$/.test(identifier)) {
+      identifierPath = `uca/1/uca-${identifier}`;
+    } else {
+      const parsedIdentifier = parseIdentifier(identifier);
+
+      identifierPath = `${parsedIdentifier[1]}/${parsedIdentifier[4]}/${parsedIdentifier[1]}-${parsedIdentifier[2]}`;
+    }
+
+    let schema = CVCSchemaLoader.local(identifierPath);
+
+    // Only load the schema remotely if a base url was provided and none was found locally
```

Comment:
Is the idea that all existing schemas will be bundle on the lib? What the size overhead? Should we include only the most use ones? What the plan for a new credential we first deploy it remote then we bundle it in a future release?

Reply:
It depends on who is using the lib. Next step is to completly remove the schemas from CC (just keep some as fictures) . The schema loaders can be provided by whoever is implementing the lib (e.g. someone else may have their own schemas)

E.g. CW can then choose to bundle schemas or not... or remotely load them

I am going to create a ticket for that

Reply:
I wonder if a `civic-schema` lib may be a good idea to be reused on civic projects?

Reply:
yeah, that sound like a good idea